### PR TITLE
feat:core: add support for recipient collateral

### PR DESCRIPTION
**Pull request by nmimran99**
feat!: allow redeem for recipient != withdrawer

This PR adds a address recipient parameter to the withdraw function in the Collateral.sol contract. It allows a user to specify a recipient other then himself.

This change is a breaking change since it requires a different call parameters to the function and also changes the event emitted from the withdraw process (Withdraw event)

### CODE REVIEW

**Review by ramenforbreakfast — APPROVED** [review]
LGTM besides some nits.

File: apps/smart-contracts/core/test/unit/Collateral.test.ts [hunk 1]
```diff
@@ -1194,9 +1220,8 @@ describe('=> Collateral', () => {
       const expectedBT = amountToWithdraw.mul(USDC_DENOMINATOR).div(parseEther('1'))
       const fee = expectedBT.mul(await collateral.getWithdrawFee()).div(FEE_DENOMINATOR)
 
-      await collateral.connect(user1).withdraw(amountToWithdraw)
-
-      expect(withdrawHook.hook).calledWith(user1.address, expectedBT, expectedBT.sub(fee))
+      await collateral.connect(user1).withdraw(recipient.address, amountToWithdraw)
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/Collateral.test.ts:1223]: nit:
```suggestion
      await collateral.connect(user1).withdraw(recipient.address, amountToWithdraw)
      
```

File: apps/smart-contracts/core/test/unit/Collateral.test.ts [hunk 2]
```diff
@@ -1156,20 +1158,44 @@ describe('=> Collateral', () => {
...
+      expect(user1.address).not.eq(recipient.address)
+      const user1BTBefore = await baseToken.balanceOf(user1.address)
+      const recipientBTBefore = await baseToken.balanceOf(recipient.address)
+      const amountToWithdraw = await collateral.balanceOf(user1.address)
+      const expectedBT = amountToWithdraw.mul(USDC_DENOMINATOR).div(parseEther('1'))
+
+      expect(recipientBTBefore).to.eq(0)
+      expect(amountToWithdraw).to.be.gt(0)
+
+      const fee = expectedBT.mul(await collateral.getWithdrawFee()).div(FEE_DENOMINATOR)
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/Collateral.test.ts:1179]:
```suggestion
      const user1BTBefore = await baseToken.balanceOf(user1.address)
      const recipientBTBefore = await baseToken.balanceOf(recipient.address)
      const amountToWithdraw = await collateral.balanceOf(user1.address)
      const expectedBT = amountToWithdraw.mul(USDC_DENOMINATOR).div(parseEther('1'))
      expect(recipientBTBefore).to.eq(0)
      expect(amountToWithdraw).to.be.gt(0)
      const fee = expectedBT.mul(await collateral.getWithdrawFee()).div(FEE_DENOMINATOR)
```

File: apps/smart-contracts/core/test/unit/Collateral.test.ts [hunk 3]
```diff
@@ -1156,20 +1158,44 @@ describe('=> Collateral', () => {
...
-      await collateral.connect(user1).withdraw(amountToWithdraw)
+      await collateral.connect(user1).withdraw(user1.address, amountToWithdraw)
 
       expect(await baseToken.balanceOf(user1.address)).to.eq(userBTBefore.add(expectedBT))
       expect(await baseToken.balanceOf(collateral.address)).to.eq(contractBTBefore.sub(expectedBT))
       expect(await collateral.balanceOf(user1.address)).to.eq(userCTBefore.sub(amountToWithdraw))
       expect(await baseToken.allowance(collateral.address, withdrawHook.address)).to.be.eq(0)
     })
 
+    it('withdraws collateral to recipient if recipient != withdrawer', async () => {
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/Collateral.test.ts:1169]: base token is withdrawn, not collateral
```suggestion
    it('withdraws to recipient if recipient != withdrawer', async () => {
```
and could you find the corresponding 'recipient = withdrawer' case and rename it to this format?

**Review by XavierEkkel — APPROVED** [review]
LGTM after nits

**Pull request merged**